StardewArchipelago/GameModifications/Buildings/BuildingMenuArchipelago.cs

```diff
@@ -41,6 +45,27 @@ protected BuildingMenuArchipelago(IModHelper modHelper, ArchipelagoClient archip
 
         public abstract List<BluePrint> GetAvailableBlueprints();
 
+        public Dictionary<string, BluePrint> FullBlueprintData()
+        {
+            var fullBlueprintData = new Dictionary<string, BluePrint>();
+            Dictionary<string, string> rawBlueprintData = Game1.content.Load<Dictionary<string, string>>("Data\\blueprints");
+            foreach (var blueprintPair in rawBlueprintData)
+            {
+                if (ExcludedBuildings.Contains(blueprintPair.Key))
+                {
+                    continue;
+                }
+                var blueprintDataArray = blueprintPair.Value.Split('/');
+                if (blueprintDataArray[0] == "animal")
+                {
+                    continue;
+                }
+                fullBlueprintData.Add(blueprintPair.Key, new BluePrint(blueprintPair.Key));

```

Comment:
I strongly dislike instantiating Stardew Blueprints here. Why not go with my previous suggestion?

Reply:
Its bloat.